xcube_stac/stac.py

```diff
@@ -157,6 +159,94 @@ def list_item_data_ids(self, items: Iterable[Item]) -> List[str]:
         """
         return list(self.get_item_data_ids(items))
 
+    def get_data_ids(
+        self,
+        items: Iterable[pystac.Item] = None,
+        item_data_ids: Iterable[str] = None,
+        include_attrs: Container[str] = None,
+        **open_params
+    ) -> Union[Iterator[str], Iterator[Tuple[str, Dict[str, Any]]]]:
+        """Get an iterator over the data resource identifiers for the
+        given type *data_type*. If *data_type* is omitted, all data
+        resource identifiers are returned. The data resource identifiers
+        follow the following structure:
+
+            `collection_id_0/../collection_id_n/item_id/asset_id`
+
+        Args:
+            items: collection of items for which data IDs are desired. If None,
+                items are collected by :meth:`get_item_collection` using *open_params*
+                Defaults to None.
+            item_data_ids: data IDs corresponding to items. If None,
+                item_data_ids are collected by :meth:`get_item_data_ids`
+                Defaults to None.
+            include_attrs: A sequence of names of attributes to be returned
+                for each dataset identifier. If given, the store will attempt
+                to provide the set of requested dataset attributes in addition
+                to the data ids. If no attributes are found, empty dictionary
+                is returned. So far only the attribute 'title' is supported.
+                Defaults to None.
+
+        Yields:
+            An iterator over the identifiers (and additional attributes defined
+            by *include_attrs* of data resources provided by this data store).
+        """
+        if items is None:
+            items, item_data_ids = self.get_item_collection(**open_params)
+        if item_data_ids is None:
+            item_data_ids = self.get_item_data_ids(items)
+
+        for (item, item_data_id) in zip(items, item_data_ids):
+            for asset in self.get_assets_from_item(
+                item, include_attrs, **open_params
+            ):
+                if include_attrs is not None:
+                    (asset, attrs) = asset
+                    data_id = (
+                        item_data_id + self._data_id_delimiter + asset
+                    )
+                    yield (data_id, attrs)
+                else:
+                    data_id = item_data_id + self._data_id_delimiter + asset
+                    yield data_id
+
+    def get_assets_from_item(
+        self,
+        item: pystac.Item,
+        include_attrs: Container[str] = None,
+        **open_params
+    ) -> Iterator[str]:
```

Comment:
To also cover the `include_attrs` case, you need:

```suggestion
    ) -> Union[Iterator[str], Iterator[Tuple[str, Mapping[str, Any]]]:
```